ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -100,7 +102,10 @@ private void putBlob(MessageInfo messageInfo, ByteBuffer messageBuf, long size)
     // May need to encrypt buffer before upload
     if (performUpload) {
       BlobId blobId = (BlobId) messageInfo.getStoreKey();
-      cloudDestination.uploadBlob(blobId, size, new ByteBufferInputStream(messageBuf));
+      CloudBlobMetadata blobMetadata =
+          new CloudBlobMetadata(blobId, messageInfo.getOperationTimeMs(), messageInfo.getSize());
+      blobMetadata.setExpirationTime(messageInfo.getExpirationTimeInMs());
```

Comment:
Why not add expirationTime to constructor?

Reply:
Fixed.